### PR TITLE
Workspace dock deletion bugs

### LINKED ISSUE #18886: Bugs with deleting workspaces from the workspace dock
A couple of issues have been spotted with workspace deletion:

 - Workspaces are deleted even if the workspace dock is not focused.
 - Workspaces are deleted without confirmation if confirmations are turned on.

### Expected behavior
 - Workspace are not deleted when the main Mantid window is not focused.
 - Workspaces are not immediately deleted when confirmations are turned on.

### Actual behavior
 - Workspace are deleted when the main Mantid window is not focused.
 - Workspaces are immediately deleted when confirmations are turned on.

### Steps to reproduce the behavior
 - A good example of the first one is trying to delete a mask from the instrument view using the delete button will actually delete the workspace instead (when it's not floating).
 - The second is obvious in that it just doesn't work.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by samueljackson92**
Fixes bugs with deleting workspaces from the Mantid dock.

**To test:**
Two things to test:
 - Try deleting a workspace using the delete key when the dock is not focused.
   - It should no longer be deleted
 - Try turning on delete confirmations (Properties -> General -> Confirmations) and check that
 you are asked to confirm before deleting a workspace.
 - Code review

Fixes #18886 .

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by PranavBahuguna** [comment]
The code looks fine, no issues to report there :+1: 

After going through the testing instructions, I found the following:
- Workspaces are no longer deleted with delete key when mantid dock is not in focus.
- While a confirmation popup is now produced when deleting workspace if 'delete confirmations' setting is checked, unchecking this setting will **not*** prevent confirmation popups from appearing on workspace deletion.

**Comment by martyngigg** [comment]
@samueljackson92 Are you looking into the issue that Pranav has found?

**Comment by samueljackson92** [comment]
@PranavBahuguna I added some code to update when the config changes. Could you please retest when you have a spare moment.

**Comment by PranavBahuguna** [comment]
On testing the new changes, unchecking the 'delete confirmations' setting will now prevent confirmation popups from appearing on workspace deletion.

An issue I still found is that if the 'delete confirmations' setting is originally unchecked on Mantid startup, confirmation popups will still appear on workspace deletion as though it was checked. I think all that is missing is a check on startup for the status of the 'delete confirmations' setting.

**Comment by martyngigg** [comment]
@samueljackson92 Did this bug exist in v3.8?

Given the level of changes and files involved I would prefer not to include this in the patch as I think the risk at impacting other things is quite high.

File: Testing/Temporary/CTestCostData.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+---
```
Comment (martyngigg) [inline Testing/Temporary/CTestCostData.txt]: I think you have committed this file by accident?

**Comment by samueljackson92** [comment]
@martyngigg understandable. I think it came in with the refactoring of the WorkspaceDock. Which wouldn't have been in 3.8

**Comment by samueljackson92** [comment]
This was moved ti #18986 and #18978

**Pull request closed without merging**